scripts/src/javatests/org/oppia/android/scripts/maven/RetrieveLicenseTextsTest.kt

```diff
@@ -0,0 +1,585 @@
+          scrapableLinkLicense
+        )
+      ),
+      getMavenDependency(
+        artifactName = "artifact.name:2",
+        artifactVersion = "2.3.1",
+        licenseList = listOf(
+          directLinkOnlyLicense
+        )
+      )
+    )
+    val mavenDependencyList = getMavenDependencyList(dependencyList)
+
+    val pbFile = tempFolder.newFile("scripts/assets/maven_dependencies.pb")
+    mavenDependencyList.writeTo(pbFile.outputStream())
+
+    RetrieveLicenseTexts(mockLicenseFetcher).main(
+      arrayOf(
+        "${tempFolder.root}/values",
+        "${tempFolder.root}/scripts/assets/maven_dependencies.pb"
+      )
+    )
+    assertThat(outContent.toString()).contains(SCRIPT_PASSED_INDICATOR)
+  }
+
+  @Test
+  fun testScript_validDependencyList_generatesResourceXmlFile() {
+    val scrapableLinkLicense = getLicenseWithScrapableLink(
+      licenseName = "Apache License",
+      originalLink = SCRAPABLE_LINK,
+      scrapableLinkUrl = SCRAPABLE_LINK
+    )
+    val directLinkOnlyLicense = getLicenseWithDirectLinkOnlyLink(
+      licenseName = "Android Terms of Service",
+      originalLink = DIRECT_LINK_ONLY,
+      directLinkOnlyUrl = DIRECT_LINK_ONLY
+    )
+    val extractedCopyLinkLicense = getLicenseWithExtractedCopyLink(
+      licenseName = "BSD License",
+      originalLink = EXTRACTED_COPY_ORIGINAL_LINK,
+      extractedCopyLinkUrl = EXTRACTED_COPY_LINK
+    )
+    val dependencyList = listOf<MavenDependency>(
+      getMavenDependency(
+        artifactName = "artifact.name:1",
+        artifactVersion = "2.0.1",
+        licenseList = listOf(scrapableLinkLicense)
+      ),
+      getMavenDependency(
+        artifactName = "artifact.name:2",
+        artifactVersion = "4.1.1",
+        licenseList = listOf(directLinkOnlyLicense, extractedCopyLinkLicense)
+      ),
+      getMavenDependency(
+        artifactName = "artifact.name:3",
+        artifactVersion = "3.1.1",
+        licenseList = listOf(scrapableLinkLicense)
+      )
+    )
+    val mavenDependencyList = getMavenDependencyList(dependencyList)
+
+    val pbFile = tempFolder.newFile("scripts/assets/maven_dependencies.pb")
+    val xmlFile = tempFolder.newFile("values/third_party_dependencies.xml")
+    mavenDependencyList.writeTo(pbFile.outputStream())
+
+    RetrieveLicenseTexts(mockLicenseFetcher).main(
+      arrayOf(
+        "${tempFolder.root}/values",
+        "${tempFolder.root}/scripts/assets/maven_dependencies.pb"
+      )
+    )
+    assertThat(outContent.toString()).contains(SCRIPT_PASSED_INDICATOR)
+    val xmlContent = xmlFile.inputStream().bufferedReader().use { it.readText() }
+    val dependencyNamesList = retrieveListOfStrings(xmlContent, "third_party_dependency_name_")
+    val dependencyVersionsList =
+      retrieveListOfStrings(xmlContent, "third_party_dependency_version_")
+    val licenseTextsList = retrieveListOfStrings(xmlContent, "license_text_")
+    val licenseNamesList = retrieveListOfStrings(xmlContent, "license_name_")
```

Comment:
Sorry, I didn't get that completely, aren't we already using helpers to retrieve a list of strings that need to be matched?

Reply:
I was thinking more of isolating the XML reading & retrieval of higher-level elements to helper methods so that all the test needs to care about is retrieving exactly the data it needs. That being said, I don't feel strongly about this.